url6da19b5a7a_krakrak_Adventure_Game_tgz-pJ8-adventure_MainJ8/src/commands/Menu.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * To change this template, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package commands;
+
+import java.util.Scanner;
+
+/**
+ * Menu of the game
+ * @author Irakiza Krak
+ */
+public class Menu extends AbstractCommand{
+
+    private int choix;
+    
+    //Constructeur
+    public Menu(){
+        super("MENU", "Shows the menu", "menu");
+    }
+
+    /**
+     *
+     * @param game
+     */
+    @Override
+        public void execute(Game game, String params) {
+            if(params.equalsIgnoreCase("quit")){
+                System.out.println("Slovensky raj : Welcome");
+                System.out.println("******************************");
+                System.out.println("Menu :");
+                System.out.println("1 - Restart Game");
+                System.out.println("2 - Quit Game");
+            
+                    
+                Scanner reader;
+                reader = new Scanner( System.in );
+                choix = reader.nextInt();
+                while((choix != 1 && choix != 2)){
+                    System.out.println("Enter a valid value");
+                    reader = new Scanner(System.in);
+                    choix = reader.nextInt();
+                }
+                if(choix == 1){
+                    game.init();
+                }
+                else if(choix == 2){
+                    Command end;
+                    end = new End();
+                    end.execute(game, null);
+                }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 44:51, 69:76.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.